extensions/amp-list/0.1/test/test-amp-list.js

```diff
@@ -239,6 +242,182 @@ describes.repeated(
             return list.layoutCallback();
           });
 
+          describe('initialized with [layout=container]', () => {
+            function expectLockedRender() {
+              // Call measureMutate before measure during render.
+              listMock
+                .expects('measureMutateElement')
+                .callsFake((m, n) => {
+                  m();
+                  n();
+                })
+                .atLeast(1);
+              listMock
+                .expects('measureElement')
+                .callsFake((m) => m())
+                .atLeast(1);
+
+              // Hide loading/placeholder during render.
+              listMock.expects('toggleLoading').withExactArgs(false).atLeast(1);
+              listMock
+                .expects('togglePlaceholder')
+                .withExactArgs(false)
+                .atLeast(1);
+            }
+
+            it('should error without experiment', () => {
+              allowConsoleError(() => {
+                expect(() => list.isLayoutSupported('container')).to.throw(
+                  'Experiment "amp-list-layout-container" is not turned on.'
+                );
+              });
+            });
+
+            describe('with experiment on', () => {
+              let itemElement;
+
+              beforeEach(() => {
+                toggleExperiment(win, 'amp-list-layout-container', true);
+                itemElement = doc.createElement('div');
+                const placeholder = doc.createElement('div');
+                placeholder.style.height = '1337px';
+                element.appendChild(placeholder);
+                element.getPlaceholder = () => placeholder;
+              });
+
+              afterEach(() => {
+                toggleExperiment(win, 'amp-list-layout-container', false);
+              });
+
+              it('should require placeholder', () => {
```

Comment:
nit: feels like there has to be a better way than duplicating all the `layout=container` tests for amp4email and feature flag cases :/

Reply:
Good point, thanks for catching this! Used `describes.repeated()` here.

Reply:
whoa, awesome diff! TIL about `describes.repeated`, super useful